services/api/src/routes/auth.js

```diff
@@ -163,21 +164,33 @@ router
       ctx.status = 204;
     }
   )
-  .post('/logout', authenticate({ type: 'user' }), fetchUser, async (ctx) => {
-    const user = ctx.state.authUser;
-
-    await AuditEntry.append('Deauthentication', ctx, {
-      object: user,
-      user: user.id,
-    });
-
-    await user.updateOne({
-      $unset: {
-        authTokenId: true,
-      },
-    });
-    ctx.status = 204;
-  })
+  .post(
+    '/logout',
+    validateBody({
+      all: Joi.boolean(),
+      jti: Joi.string(),
+    }),
+    authenticate({ type: 'user' }),
+    fetchUser,
+    async (ctx) => {
+      const user = ctx.state.authUser;
+      const { body } = ctx.request;
+      const jti = body.jti || ctx.state.jwt.jti;
+      if (body.all) {
+        user.authTokens = [];
+      } else if (jti) {
+        user.authTokens = user.authTokens.filter((token) => token.jti !== jti);
```

Comment:
it kind of seems like if there's a `createAuthToken` method then this belongs in a `removeAuthToken` method as well